### PR TITLE
Basic support for transactions

### PR DESCRIPTION
**Pull request by pimeys**
With this change, it is possible to start, commit and rollback
transactions using the `simple_query` method in the `Client`. Querying
`BEGIN TRAN` will start a new transaction and store the current
transaction descriptor, adding it to the subsequent queries until
`COMMIT` or `ROLLBACK` happens.

Please be careful on always commiting or rolling back before subsequent
queries to always keep the transaction descriptor in a correct value.

This commit is a preliminary setup for allowing to use transactions.
Transaction management and automatic rollbacks should be implemented in
a separate pull request.

### CODE REVIEW

**Comment by pimeys** [comment]
How this works:

When querying `BEGIN TRAN`, the stream will have an environment change token containing an id for the transaction. All subsequent RPC and batch messages must use this descriptor. In case of `COMMIT` or `ROLLBACK`, the server responds with an environment change token that then sets the current transaction id to 0.

File: src/tds.rs [hunk 1]
```diff
@@ -78,8 +80,20 @@ impl Context {
...
+    pub fn set_packet_size(&self, new_size: u32) {
+        self.packet_size.store(new_size, Ordering::SeqCst);
+    }
+
+    pub fn transaction_id(&self) -> u64 {
+        self.transaction_id.load(Ordering::SeqCst)
+    }
+
+    pub fn set_transaction_id(&self, id: u64) {
+        self.transaction_id.store(id, Ordering::SeqCst);
```
Comment (tomhoule) [inline src/tds.rs:96]: My only worry is that here multiple threads could be setting the transaction id on the client at the same time. But seeing token_env_change.rs it looks like this piece of state is server-driven and per-connection, right?

Comment (tomhoule) [reply src/tds.rs:96]: Like you said in your previous comment.

Comment (pimeys) [reply src/tds.rs:96]: The atomics are here only for ergonomic reasons. A connection is always single-threaded in every SQL database, so there is no way of threads being able to mess with the ids.

As you see the `Client` takes `&mut self` always, which forces this to be single-threaded (or at least sync with a `Mutex`).

**Pull request merged**